aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -926,21 +880,67 @@ std::tuple<at::Tensor, at::Tensor> _efficient_attention_forward(
   at::Tensor res;
   at::Tensor logsumexp;
 
-  auto launchKernel = [&](auto _k, int computeCapability) {
+  const bool use_dropout = std::fpclassify(dropout_p) != FP_ZERO;
+  at::PhiloxCudaState rng_engine_inputs;
+  if (use_dropout) {
+    at::CUDAGeneratorImpl* gen =
+        at::get_generator_or_default<at::CUDAGeneratorImpl>(
+            c10::nullopt, at::cuda::detail::getDefaultCUDAGenerator());
+
+    std::lock_guard<std::mutex> lock(gen->mutex_);
+    // if using dropout, we produce 1 random number for each element of the
+    // attention tensor
+    rng_engine_inputs = gen->philox_cuda_state(B * num_heads * M * N);
```

Comment:
This will need to change folllow this pattern: https://github.com/pytorch/pytorch/pull/100196

I think since we are adding dropout and bias in subsequent PRs it is okay to also do these changes in follow ups